lsp/nls/src/requests/completion.rs

```diff
@@ -27,20 +28,21 @@ fn remove_duplicates_and_myself(
     items: &[CompletionItem],
     cursor: RawPos,
 ) -> Vec<lsp_types::CompletionItem> {
-    let mut seen_labels = HashSet::new();
-    let mut ret = Vec::new();
+    let mut grouped = HashMap::<String, CompletionItem>::new();
     for item in items {
         if let Some(ident) = item.ident {
             if ident.pos.contains(cursor) {
                 continue;
             }
         }
 
-        if seen_labels.insert(&item.label) {
-            ret.push(item.clone().into());
-        }
+        grouped
+            .entry(item.label.clone())
+            .and_modify(|old| *old = Combine::combine(old.clone(), item.clone()))
```

Comment:
Could be nice to write documentation for this function explaining that it combines the metadata of identical items, because I fear it's not obvious from the function's name anymore

Reply:
I refactored it a little to make it more clear (and also added documentation)